osf_tests/test_analytics.py

```diff
@@ -42,3 +44,132 @@ def test_increment_user_activity_counters(self):
+@pytest.fixture()
+def page_counter_for_individual_version(project, file_node3):
+    page_counter_id = 'download:{}:{}:0'.format(project._id, file_node3.id)
+    page_counter, created = PageCounter.objects.get_or_create(_id=page_counter_id, date={u'2018/02/04': {u'total': 1, u'unique': 1}})
+    return page_counter
+
+
+class TestPageCounter:
+
+    @mock.patch('osf.models.analytics.session')
+    def test_download_update_counter(self, mock_session, project, file_node):
+        mock_session.data = {}
+        page_counter_id = 'download:{}:{}'.format(project._id, file_node.id)
+
+        PageCounter.update_counter(page_counter_id, {})
+
+        page_counter = PageCounter.objects.get(_id=page_counter_id)
+        assert page_counter.total == 1
+        assert page_counter.unique == 1
+
+        PageCounter.update_counter(page_counter_id, {})
+
+        page_counter.refresh_from_db()
+        assert page_counter.total == 2
+        assert page_counter.unique == 1
+
+    @mock.patch('osf.models.analytics.session')
+    def test_download_update_counter_contributor(self, mock_session, user, project, file_node):
+        mock_session.data = {'auth_user_id': user._id}
+        page_counter_id = 'download:{}:{}'.format(project._id, file_node.id)
+
+        PageCounter.update_counter(page_counter_id, {'contributors': project.contributors})
+        page_counter = PageCounter.objects.get(_id=page_counter_id)
+        assert page_counter.total == 0
+        assert page_counter.unique == 0
+
+        PageCounter.update_counter(page_counter_id, {'contributors': project.contributors})
+
+        page_counter.refresh_from_db()
+        assert page_counter.total == 0
+        assert page_counter.unique == 0
+
+    def test_get_all_downloads_on_date(self, page_counter, page_counter2):
+        """
+        This method tests that multiple pagecounter objects have their download totals summed properly.
+
+        :param page_counter: represents a page_counter for a file node being downloaded
+        :param page_counter2: represents a page_counter for another file node being downloaded
+        """
+
+        date = datetime(2018, 2, 4)
+
+        total_downloads = PageCounter.get_all_downloads_on_date(date)
+
+        assert total_downloads == 45
+
+    def test_get_all_downloads_on_date_exclude_versions(self, page_counter, page_counter2, page_counter_for_individual_version):
+        """
+        This method tests that individual version counts for file node's aren't "double counted" in the totals
+        for a page counter. We don't add the file node's total to the versions total.
+
+        :param page_counter: represents a page_counter for a file node being downloaded
+        :param page_counter2: represents a page_counter for another file node being downloaded
+        """
+        date = datetime(2018, 2, 4)
+
+        total_downloads = PageCounter.get_all_downloads_on_date(date)
+
+        assert total_downloads == 45
+
+
+class TestPageCounterRegex:
```

Comment:
These will still use a database because this whole file is marked with

```
pytestmark = pytest.mark.django_db
```

You'll need to explicitly mark the classes that need a db and keep this one unmarked.